lib/matplotlib/tri/tricontour.py

```diff
@@ -103,175 +108,222 @@ def _contour_args(self, args, kwargs):
+        colormap are changed. Therefore, an explicit call to
+        `.ContourSet.changed()` is needed after modifying the colormap. The
+        explicit call can be left out, if a colorbar is assigned to the
+        `.TriContourSet` because it internally calls `.ContourSet.changed()`.
 
-        If contour is using a monochrome colormap and the contour
-        level is less than 0, then the linestyle specified
-        in :rc:`contour.negative_linestyle` will be used.
+xunits, yunits : registered units, optional
+    Override axis units by specifying an instance of a
+    :class:`matplotlib.units.ConversionInterface`.""")
 
-    tricontourf-only keyword arguments:
 
-        *antialiased*: bool
-        enable antialiasing
+@docstring.dedent_interpd
+def tricontour(ax, *args, **kwargs):
+    """
+    %(_tricontour_doc)s
 
-    Note: `.tricontourf` fills intervals that are closed at the top; that is,
-    for boundaries *z1* and *z2*, the filled region is::
+    linewidths : float or array-like, default: :rc:`contour.linewidth`
+        The line width of the contour lines.
 
-        z1 < Z <= z2
+        If a number, all levels will be plotted with this linewidth.
 
-    except for the lowest interval, which is closed on both sides (i.e. it
-    includes the lowest value).
+        If a sequence, the levels in ascending order will be plotted with
+        the linewidths in the order specified.
+
+        If None, this falls back to :rc:`lines.linewidth`.
+
+    linestyles : {*None*, 'solid', 'dashed', 'dashdot', 'dotted'}, optional
+        If *linestyles* is *None*, the default is 'solid' unless the lines are
+        monochrome.  In that case, negative contours will take their linestyle
+        from :rc:`contour.negative_linestyle` setting.
+
+        *linestyles* can also be an iterable of the above strings specifying a
+        set of linestyles to be used. If this iterable is shorter than the
+        number of contour levels it will be repeated as necessary.
     """
     kwargs['filled'] = False
     return TriContourSet(ax, *args, **kwargs)
+if tricontour.__doc__:
+    tricontour.__doc__ = tricontour.__doc__ % {
+        'func': 'tricontour',
+        'type': 'lines',
+    }
 
 
+@docstring.dedent_interpd
 def tricontourf(ax, *args, **kwargs):
-    kwargs['filled'] = True
-    return TriContourSet(ax, *args, **kwargs)
+    """
+    %(_tricontour_doc)s
+
+    antialiased : bool, default True
+        Enable antialiasing, overriding the defaults.
 
+    Notes
+    -----
+    `.tricontourf` fills intervals that are closed at the top; that is, for
+    boundaries *z1* and *z2*, the filled region is::
 
-tricontourf.__doc__ = tricontour.__doc__
+        z1 < Z <= z2
+
+    except for the lowest interval, which is closed on both sides (i.e. it
+    includes the lowest value).
+    """
+    kwargs['filled'] = True
+    return TriContourSet(ax, *args, **kwargs)
+if tricontourf.__doc__:
```

Comment:
Can we have a case where that is necessary? And do our other docstring manipulation routines check that as well?

Reply:
Yes, when running with optimization. The functions in `docstring` and `pyplot` do check for this.

Reply:
Actually, I came up with a simpler way to do this, while looking at other docs.